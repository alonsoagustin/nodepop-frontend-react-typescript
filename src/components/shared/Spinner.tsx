interface Props {
  isLoading: boolean;
}
const Spinner = ({ isLoading }: Props) => {
  if (!isLoading) return null;

  return (
    <div className="d-flex justify-content-center">
      <div className="spinner-border text-primary" role="status">
        <span className="visually-hiden">Loading...</span>
      </div>
    </div>
  );
};

export default Spinner;
