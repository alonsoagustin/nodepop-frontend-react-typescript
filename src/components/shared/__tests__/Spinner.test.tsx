import { render } from "@testing-library/react";
import Spinner from "../Spinner";

describe("Spinner", () => {
  test("should render spinner", () => {
    const { container } = render(<Spinner isLoading />);
    expect(container).toMatchSnapshot("spinner-loading");
  });

  test("should not render spinner", () => {
    const { container } = render(<Spinner isLoading={false} />);
    expect(container).toMatchSnapshot("spinner-not-loading");
  });
});
