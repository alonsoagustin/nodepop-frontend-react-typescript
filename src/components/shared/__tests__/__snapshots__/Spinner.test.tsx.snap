// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Spinner > should not render spinner > spinner-not-loading 1`] = `<div />`;

exports[`Spinner > should render spinner > spinner-loading 1`] = `
<div>
  <div
    class="d-flex justify-content-center"
  >
    <div
      class="spinner-border text-primary"
      role="status"
    >
      <span
        class="visually-hidden"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;
